File: java/darts/src/main/java/Darts.java
class Darts {

    private final int score;

    Darts(double x, double y) {
        double dist = Math.hypot(x, y);

        if (dist > 10)
            this.score = 0;
        else if (dist > 5)
            this.score = 1;
        else if (dist > 1)
            this.score = 5;
        else
            this.score = 10;
    }

    int score() {
        return this.score;
    }

}

File: c/darts/src/darts.c
#include "darts.h"
#include <math.h>

int score(coordinate_t c)
{
	double dist = hypot(c.x, c.y);

	return dist > 10 ? 0
		  : dist > 5 ? 1
		  : dist > 1 ? 5
					 : 10;
}
